buildSrc/src/test/kotlin/com/datadog/gradle/plugin/jsonschema/TestDefinitions.kt

```diff
@@ -118,6 +118,49 @@ val Comment = TypeDefinition.Class(
     )
 )
 
+val Company = TypeDefinition.Class(
+    name = "Company",
+    properties = listOf(
+        TypeProperty("name", TypeDefinition.Primitive(JsonPrimitiveType.STRING), true),
+        TypeProperty(
+            "ratings",
+            TypeDefinition.Class(
+                name = "Ratings",
+                properties = listOf(
+                    TypeProperty(
+                        "global",
+                        TypeDefinition.Primitive(JsonPrimitiveType.INTEGER),
+                        false
+                    )
+                ),
+                additionalProperties = TypeDefinition.Primitive(JsonPrimitiveType.INTEGER)
```

Comment:
so per JSON definition if `additionalProperties` is a of primitive type, then it maps to primitive at the class level, but if it is `object` we try to map it to `Map<String, Any?>`, because this is language representation of `JSONObject`. Do I get it right?

Reply:
Yes, we had to use this trick to support this `Map<String,*>` additionalProperties attribute. The reason why `Object` maps to `Map<String,Any?>` is that our models are inferred from the Public API in a certain way and the `additionalProperties` means something that the user can send from its end and we don't impose a specific type sometimes (from there we get the `Any` value). The reason why we have optionals (`Any?` instead of `Any`) is because that Public API can be called from a Java code and there's no way we can force usage of non nullable values in there, therefore we need to be prepared to handle those case on our end.

Reply:
makes sense :) so it is freeform user metadata.